NYTPhotoViewer/NYTPhotoTransitionAnimator.m

```diff
@@ -120,7 +120,13 @@ - (void)performZoomingAnimationWithTransitionContext:(id <UIViewControllerContex
     if (!endingViewForAnimation) {
         endingViewForAnimation = [[self class] newAnimationViewFromView:self.endingView];
     }
-    
+
+    if (!startingViewForAnimation || !endingViewForAnimation) {
+        NSAssert(startingViewForAnimation && endingViewForAnimation, @"When starting or ending views are nil, zooming should not be performed.");
+        [self completeTransitionWithTransitionContext:transitionContext];
```

Comment:
Yes: When we return early, we must still signal that the transition should complete. Else we may be stuck somewhere forever.